src/SmartTransactionsController.ts

```diff
@@ -889,4 +889,56 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       );
     });
   }
+
+  #wipeSmartTransactionsPerChainId(
+    chainId: Hex,
+    selectedAddressLowerCase: string,
+  ): void {
+    const { smartTransactions } = this.state.smartTransactionsState;
+    const smartTransactionsForSelectedChain: SmartTransaction[] =
+      smartTransactions?.[chainId];
+    if (
+      !smartTransactionsForSelectedChain ||
+      smartTransactionsForSelectedChain.length === 0
+    ) {
+      return;
+    }
+    const newSmartTransactionsForSelectedChain =
+      smartTransactionsForSelectedChain.filter(
+        (smartTransaction: SmartTransaction) =>
+          smartTransaction.txParams?.from !== selectedAddressLowerCase,
+      );
+    this.update({
+      smartTransactionsState: {
+        ...this.state.smartTransactionsState,
+        smartTransactions: {
+          ...smartTransactions,
+          [chainId]: newSmartTransactionsForSelectedChain,
+        },
+      },
+    });
+  }
+
+  wipeSmartTransactions(
+    selectedAddress?: string,
+    ignoreNetwork?: boolean,
+  ): void {
+    if (!selectedAddress) {
+      return;
+    }
+    const selectedAddressLowerCase = selectedAddress.toLowerCase();
+    if (ignoreNetwork) {
```

Comment:
Does using this condition and helper method here complicate the method a little and make it more rigid if additional filters are added in future?

Would it be simpler and more extensible to have a single loop going through a flattened list of all smart transactions and filtering as needed, then the map can be reassembled at the end with a `#groupTransactionsByChainId` for example?

You could even use a helper to get the flattened list such as `#getAllSmartTransactions()`.

Reply:
Which additional filters are you thinking about? I don't think this function will change anytime soon and wiping smart transactions per chainId seems simple enough.